### ISSUE TITLE
Mesh is missing parts

### ISSUE THREAD

**Issue by tonpix** [f3d-app/f3d#1675]
**Describe the bug**
Exported as .usdc from blender 4.4, then imported into blender and opened in F3D for comparison. Mesh in F3D is missing some parts - look at screenshot.
![f3d_fa3eQp2vIq](https://github.com/user-attachments/assets/35830bde-14cc-42ff-83dd-fea0eb7a35f0)

**To Reproduce**
1. open roof.usdc

**Expected behavior**
Mesh should be identical in blender and F3D

**System Information:**
 - OS: Windows 10
 - GPU and GPU driver: GTX 1080, drivers 551.76

**F3D Information**
Version: 2.5.0.
Build date: 2024-07-01 21:30:06.
Build system: Windows 64-bits.
Compiler: MSVC 19.40.33811.0.
External rendering module: ON.
Raytracing module: ON.
VTK version: 9.3.0-3457-gecf3646d64 (date: 20240620).

**Additional context**
Attached roof.usdc
[roof.zip](https://github.com/user-attachments/files/17490598/roof.zip)

**Comment by Meakk** [comment]
There is a GeomSubset under the `Plane_042` mesh node. It defines a submesh by selecting only a few faces. When this GeomSubset node is removed, it works well:
<img width="499" alt="image" src="https://github.com/user-attachments/assets/6b44b9b8-62c5-459e-9bf4-f217a8611836">

geomsubset removed: [roof.zip](https://github.com/user-attachments/files/17491217/roof.zip)

I don't know what's the expected behavior though.

**Comment by tonpix** [comment]
> There is a GeomSubset under the `Plane_042` mesh node. It defines a submesh by selecting only a few faces. When this GeomSubset node is removed, it works well: <img alt="image" width="499" src="https://private-user-images.githubusercontent.com/6069752/379266112-6b44b9b8-62c5-459e-9bf4-f217a8611836.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk2ODk3MTQsIm5iZiI6MTcyOTY4OTQxNCwicGF0aCI6Ii82MDY5NzUyLzM3OTI2NjExMi02YjQ0YjliOC02MmM1LTQ1OWUtOWJmNC1mMjE3YTg2MTE4MzYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTAyMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEwMjNUMTMxNjU0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzMzYzA5MjBmMmIxM2EyNDU0NzA1YzBjYjliYTM5OWMzNjkwOWVkMGNiMjg2YWE5NmYzY2UxYWY4OGIwNDFkZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.0sVGJZqtAACULRmz9dZFCiS1mvOGl5ikX1uhiqx90zk">
> 
> geomsubset removed: [roof.zip](https://github.com/user-attachments/files/17491217/roof.zip)
> 
> I don't know what's the expected behavior though.

How did you delete GeomSubset?

Expected behaviour is that loading .usdc file into F3D would show everything, just like when importing same .usdc file into blender or Unreal Engine - both blender and UE5 show missing mesh parts without any modifications.

**Comment by Meakk** [comment]
> How did you delete GeomSubset?

I used Omiverse Create.
Where is your file exported from?

**Comment by tonpix** [comment]
blender 4.4. I noticed also that if I delete custom split normals data in blender and export to .usdc, F3D shows mesh correctly

**Comment by Meakk** [comment]
Related code: https://github.com/f3d-app/f3d/blob/38054ab51a4a6bf4a42cb482aefc74ee36b57009/plugins/usd/module/vtkF3DUSDImporter.cxx#L661-L698

I don't remember why I did that, but I think it was needed. Maybe it's a Blender exporter issue?

**Comment by Meakk** [comment]
Ok I do remember now. It was implemented to fix https://github.com/f3d-app/f3d/issues/1186 where materials were assigned per geomsubset. But in this case, the subsets combined make the whole mesh, which is not the case for you.  
It's still unclear if it's a Blender issue, or if it's expected that the subsets cover only a part of the mesh.

**Comment by Meakk** [comment]
Yes I tried too. It seems like the important parameter is `subsetFamily` on the mesh:

<img width="869" alt="image" src="https://github.com/user-attachments/assets/f6cd3c53-181a-4d5b-9ac7-7541485ac12d">

In the [USD documentation](https://openusd.org/release/api/class_usd_geom_subset.html), I can read:

> When multiple subsets belonging to a prim have the same familyName, they are said to belong to the family.
> A familyType value can be encoded on the owner of a family of subsets as a token using the static method [UsdGeomSubset::SetFamilyType()](https://openusd.org/release/api/class_usd_geom_subset.html#ad0c186e761984f80850e2e820f5ac6f9). "familyType" can have one of the following values:
> **UsdGeomTokens->partition**: implies that every element of the whole geometry appears exactly once in only one of the subsets belonging to the family.
> **UsdGeomTokens->nonOverlapping**: an element that appears in one subset may not appear in any other subset belonging to the family, and appears only once in the subset in which it appears.
> **UsdGeomTokens->unrestricted**: implies that there are no restrictions w.r.t. the membership of elements in the subsets. They could be overlapping and the union of all subsets in the family may not represent the whole.

I guess `partition` is the current behavior (loop on subsets and draw them individually).  
But what about `nonOverlapping`? Display subsets and keep track of elements excluded from subsets to draw them in a single actor?  
And `unrestricted`? Looks odd to me, so maybe we can just skip that and draw the whole mesh?
